src/jit/flowgraph.cpp

```diff
@@ -13769,6 +13769,13 @@ bool Compiler::fgOptimizeSwitchBranches(BasicBlock* block)
         GenTree* switchVal = switchTree->gtOp.gtOp1;
         noway_assert(genActualTypeIsIntOrI(switchVal->TypeGet()));
 
+        // If we are in LIR, remove the switch table from the block.
+        if (block->IsLIR())
```

Comment:
This comment is confusing, because we've asserted above "assert(switchTree->OperGet() == GT_SWITCH_TABLE);". You should probably assert what it is you expect to find at gtOp2. Is it GT_JMPTABLE?